app/shop.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int, int],
+            products: dict[str: int | float]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+        self.each_product_cost = {}
+        self.total_cost = 0
+
+    def purchase(self, product_cart: dict) -> int:
+        for product in product_cart:
+            try:
+                product_cost = product_cart[product] * self.products[product]
+                if (isinstance(product_cost, float)
+                        and product_cost.is_integer()):

```

Comment:
The condition `isinstance(product_cost, float) and product_cost.is_integer()` is used to check if a float is actually an integer value. This is correct and ensures that the product cost is converted to an integer if applicable.